File: src/rpyd_client.py
#!/usr/bin/env python
import sys
from rpy import r

from rpyd import rpyd
from rpyd.ttypes import Response, Predictor

from thrift import Thrift
from thrift.transport import TSocket
from thrift.transport import TTransport
from thrift.protocol import TBinaryProtocol

if __name__ == "__main__":
  rpyd_server = 'localhost'
  rpyd_port = 9111

  transport = TSocket.TSocket(rpyd_server, rpyd_port)
  transport = TTransport.TBufferedTransport(transport)
  protocol = TBinaryProtocol.TBinaryProtocol(transport)

  client = rpyd.Client(protocol)

  transport.open()

  Y = Response({"name": "response",
                "data": range(3)})

  x0 = Predictor({"name": "x0"})
  x0.data = range(3)

  x1 = Predictor()
  x1.name = "x1"
  x1.data = [1, 1, 2]

  X = [x0, x1]

  model = client.lm(Y, X)
  print "Client: " + str(model)

  transport.close()

File: src/rpyd_client.php
<?php

include_once // generated rpyd.php file
include_once // path_to_thrift.'/protocol/TBinaryProtocol.php';
include_once // path_to_thrift.'/transport/TBufferedTransport.php';
include_once // path_to_thrift.'/transport/TSocket.php';

function rpyd_get_rpyd_client() {
  try {
    $rpyd_server = 'localhost';
    $rpyd_port = 9111;

    $sock = new TSocket($rpyd_server, $rpyd_port);
    $trans = new TBufferedTransport($sock);
    $prot = new TBinaryProtocol($trans);

    // Create the client
    $rpyd_client = new rpydClient($prot);

    // Open the transport (we rely on PHP to close it at script termination)
    $trans->open();

    // return rpyd client
    return $rpyd_client;
  } catch (Exception $x) {
    echo 'you should handle this guy.';
  }
}

function rpyd_create_response($name, $data) {
  $resp = new Response;
  $resp->name = $name;
  $resp->data = $data;
  return $resp;
}

function rpyd_create_predictor($name, $data, $is_factor = 0) {
  $pred = new Predictor;
  $pred->name = $name;
  $pred->data = $data;
  $pred->is_factor = $is_factor;
  return $pred;
}

function rpyd_test_client() {
  $rpyd_client = rpyd_get_rpyd_client();
  $Y = rpyd_create_response('Y', array(0, 1, 2));
  $x0 = rpyd_create_predictor('x0', array(0, 1, 2));
  $x1= rpyd_create_predictor('x1', array(1, 1, 2));
  $X = array($x0, $x1);
  $mdl = $rpyd_client->lm($Y, $X);
  print_r($mdl);
}
